src/util/vehicle.hpp

```diff
@@ -704,4 +928,90 @@ namespace big::vehicle
+				}
+				if (VEHICLE::GET_IS_VEHICLE_SECONDARY_COLOUR_CUSTOM(pedVeh))
+				{
+					int r, g, b;
+					VEHICLE::GET_VEHICLE_CUSTOM_SECONDARY_COLOUR(pedVeh, &r, &g, &b);
+					VEHICLE::SET_VEHICLE_CUSTOM_SECONDARY_COLOUR(playerVeh, r, g, b);
+				}
+				if (VEHICLE::IS_THIS_MODEL_A_CAR(vehicleModelHash) || VEHICLE::IS_THIS_MODEL_A_BIKE(vehicleModelHash))
+				{
+					VEHICLE::SET_VEHICLE_MOD_KIT(playerVeh, 0);
+					VEHICLE::SET_VEHICLE_WHEEL_TYPE(playerVeh, VEHICLE::GET_VEHICLE_WHEEL_TYPE(pedVeh));
+					for (int i = 0; i <= 24; i++)
+					{
+						if (i > 16 && i < 23)
+							VEHICLE::TOGGLE_VEHICLE_MOD(playerVeh, i, VEHICLE::IS_TOGGLE_MOD_ON(pedVeh, i));
+						else
+							VEHICLE::SET_VEHICLE_MOD(playerVeh, i, VEHICLE::GET_VEHICLE_MOD(pedVeh, i), VEHICLE::GET_VEHICLE_MOD_VARIATION(pedVeh, i));
+					}
+					int tireSmokeColor[3], pearlescentColor, wheelColor;
+					VEHICLE::GET_VEHICLE_TYRE_SMOKE_COLOR(pedVeh, &tireSmokeColor[0], &tireSmokeColor[1], &tireSmokeColor[2]);
+					VEHICLE::SET_VEHICLE_TYRE_SMOKE_COLOR(playerVeh, tireSmokeColor[0], tireSmokeColor[1], tireSmokeColor[2]);
+					VEHICLE::SET_VEHICLE_NUMBER_PLATE_TEXT_INDEX(playerVeh, VEHICLE::GET_VEHICLE_NUMBER_PLATE_TEXT_INDEX(pedVeh));
+					VEHICLE::SET_VEHICLE_NUMBER_PLATE_TEXT(playerVeh, VEHICLE::GET_VEHICLE_NUMBER_PLATE_TEXT(pedVeh));
+					VEHICLE::GET_VEHICLE_EXTRA_COLOURS(pedVeh, &pearlescentColor, &wheelColor);
+					VEHICLE::SET_VEHICLE_EXTRA_COLOURS(playerVeh, pearlescentColor, wheelColor);
+					if (VEHICLE::IS_VEHICLE_A_CONVERTIBLE(pedVeh, 0))
+					{
+						int convertableState = VEHICLE::GET_CONVERTIBLE_ROOF_STATE(pedVeh);
+						if (convertableState == 0 || convertableState == 3 || convertableState == 5)
+							VEHICLE::RAISE_CONVERTIBLE_ROOF(playerVeh, 1);
+						else
+							VEHICLE::LOWER_CONVERTIBLE_ROOF(playerVeh, 1);
+					}
+					for (int i = 0; i <= 3; i++)
+					{
+						VEHICLE::SET_VEHICLE_NEON_ENABLED(playerVeh, i, VEHICLE::GET_VEHICLE_NEON_ENABLED(pedVeh, i));
+					}
+					for (int i = 0; i <= 11; i++)
+					{
+						if (VEHICLE::DOES_EXTRA_EXIST(pedVeh, i))
+							VEHICLE::SET_VEHICLE_EXTRA(playerVeh, i, ~VEHICLE::IS_VEHICLE_EXTRA_TURNED_ON(pedVeh, i));
+					}
+					if ((VEHICLE::GET_VEHICLE_LIVERY_COUNT(pedVeh) > 1) && VEHICLE::GET_VEHICLE_LIVERY(pedVeh) >= 0)
+					{
+						VEHICLE::SET_VEHICLE_LIVERY(playerVeh, VEHICLE::GET_VEHICLE_LIVERY(pedVeh));
+					}
+					int neonColor[3];
+					VEHICLE::GET_VEHICLE_NEON_COLOUR(pedVeh, &neonColor[0], &neonColor[1], &neonColor[2]);
+					VEHICLE::SET_VEHICLE_NEON_COLOUR(playerVeh, neonColor[0], neonColor[1], neonColor[2]);
+					VEHICLE::SET_VEHICLE_WINDOW_TINT(playerVeh, VEHICLE::GET_VEHICLE_WINDOW_TINT(pedVeh));
+					VEHICLE::SET_VEHICLE_DIRT_LEVEL(playerVeh, VEHICLE::GET_VEHICLE_DIRT_LEVEL(pedVeh));
+				}
+				STREAMING::SET_MODEL_AS_NO_LONGER_NEEDED(vehicleModelHash);
+
+				if (*g_pointers->m_is_session_started)
+				{
+					vehicle::set_mp_bitset(vehicleModelHash);
+				}
+			}
+		}
+	}
```

Comment:
I feel like this can be replicated by the following functions found in `src/views/vehicle/view_spawn_vehicle.cpp`

https://github.com/YimMenu/YimMenu/blob/f360d7f4366d9a591f1d1ee6c26e5bdf52fcdfb6/src/views/vehicle/view_spawn_vehicle.cpp#L89-L91

Reply:
the method I'm using right now works perfectly fine